packages/docusaurus-migrate/src/index.ts

```diff
@@ -0,0 +1,707 @@
+  } catch (errorInStatic) {
+    console.log(
+      chalk.red(`Error occurred while copying static folder: ${errorInStatic}`),
+    );
+  }
+  try {
+    migrateBlogFiles(siteDir, newDir, classicPreset, shouldMigrateMdFiles);
+  } catch (errorInMigratingBlogs) {
+    console.log(
+      chalk.red(
+        `Error occurred while migrating blogs: ${errorInMigratingBlogs}`,
+      ),
+    );
+  }
+  try {
+    handleVersioning(siteDir, newDir, config, shouldMigrateMdFiles);
+  } catch (errorInVersion) {
+    console.log(
+      chalk.red(
+        `Error occurred while migrating versioned docs: ${errorInVersion}`,
+      ),
+    );
+  }
+
+  try {
+    migrateLatestDocs(siteDir, newDir, shouldMigrateMdFiles);
+  } catch (errorInDoc) {
+    chalk.red(`Error occurred while migrating docs: ${errorInDoc}`);
+  }
+
+  try {
+    migrateLatestSidebar(siteDir, newDir, classicPreset, siteConfig);
+  } catch (error) {
+    console.log(chalk.red(`Error occurred while migrating sidebar: ${error}`));
+  }
+
+  try {
+    fs.writeFileSync(
+      path.join(newDir, 'docusaurus.config.js'),
+      `module.exports=${JSON.stringify(config, null, 2)}`,
+    );
+    console.log(
+      chalk.green(
+        `Successfully created a new config file with new navbar and footer config`,
+      ),
+    );
+  } catch (error) {
+    console.log(
+      chalk.red(`Error occurred while creating config file: ${error}`),
+    );
+  }
+  try {
+    migratePackageFile(siteDir, deps, newDir);
+  } catch (error) {
+    console.log(
+      chalk.red(
+        `Error occurred while creating package.json file for project: ${error}`,
+      ),
+    );
+  }
+  console.log('Completed migration from v1 to v2');
+}
+
+export function createConfigFile(
+  siteConfig: VersionOneConfig,
+): VersionTwoConfig {
+  const homePageId = siteConfig.headerLinks?.filter((value) => value.doc)[0]
+    .doc;
+
+  const customConfigFields: Record<string, any> = {};
+  // add fields that are unknown to v2 to customConfigFields
+  Object.keys(siteConfig).forEach((key: any) => {
+    const knownFields = [
+      'title',
+      'tagline',
+      'url',
+      'baseUrl',
+      'organizationName',
+      'projectName',
+      'scripts',
+      'stylesheets',
+      'favicon',
+    ];
+    const value = siteConfig[key as keyof typeof siteConfig];
+    if (value !== undefined && !knownFields.includes(key)) {
+      customConfigFields[key] = value;
+    }
+  });
+
+  const result: VersionTwoConfig = {
+    title: siteConfig.title ?? '',
+    tagline: siteConfig.tagline,
+    url: siteConfig.url ?? '',
+    baseUrl: siteConfig.baseUrl ?? '',
+    organizationName: siteConfig.organizationName,
+    projectName: siteConfig.projectName,
+    scripts: siteConfig.scripts,
+    stylesheets: siteConfig.stylesheets,
+    favicon: siteConfig.favicon ?? '',
+    customFields: customConfigFields,
+    presets: [
+      [
+        '@docusaurus/preset-classic',
+        {
+          docs: {
+            path: 'docs',
```

Comment:
I'd like that we don't "copy" the docs, instead we should modify them directly in their original folder. They'd preserve their git history + user would be able to inspect the diff to see what has changed.

You can do for example:

```
path.relative(newSiteDir, siteDir),
const docsPath = path.join(path.relative(newSiteDir, path.join(siteDir,..)),"docs");
```

You'd get something like path: "../docs"

(not sure it works but it should)

Reply:
for some unknown reason docs do not work out of website folder

Reply:
done